src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc

```diff
@@ -258,8 +258,10 @@ void FiniteDiffGroundTempsModel::getWeatherData()
         NumOfWarmupDays = 0;
 
         annualAveAirTemp_num = 0.0;
+        // Protect against array bounds error
+        int maxSimDays = min(NumDaysInYear, NumOfDayInEnvrn);
 
-        while ((DayOfSim < NumOfDayInEnvrn) || (WarmupFlag)) { // Begin day loop ...
+        while ((DayOfSim < maxSimDays) || (WarmupFlag)) { // Begin day loop ...
```

Comment:
@jasondegraw This was failing here with an array bounds exceeded, because `NumOfDayInEnvrn` was 366 for test file PipingSystem_Underground_TwoPipe_FD_GroundTemps. That's still the case here, but this will run now. I will include this in the followup issue.
@mitchute This section is suspect, because it's looping on `NumOfDayInEnvrn`, but the average temperature calc [here](https://github.com/NREL/EnergyPlus/blob/c4de98c52a99dccf8ef85e34b9a50a5fbda0f0aa/src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc#L350) is dividing by `NumDaysInYear`. Seems like it should be dividing by `DayOfSim`

Reply:
```DayOfSim``` should work. It doesn't look to me like this is being reset after the loop, so that should still be valid.